### ISSUE TITLE
TA comments 

### ISSUE THREAD

**Issue by ChadFibke** [STAT545-UBC-students/hw05-zjbthomas#1]
# **Coding Style**

### *The good* :thumbsup:

* Good amount of code chunk comments to explain more complex code! 

* Code is readable, nicely spaced, and each layer serves its on purpose!

### *Needs work* :thumbsdown:

* Please use more informative object and image names when your saving files! It's a pain in the neck when you have a lot of random objects in your environment, and you need to sift through them all to see which one you need. 

# **Coding Strategy**

### *The good* :thumbsup:

* Broke the challenges in the manageable bite size pieces by using consistent markdown syntax.

* Good amount of in line code introducing the following code chunk. This helps inform the reader on what is coming next, and ultimately increases the flow!

* Good amount of commenting on the output! This will help guide future-you, or other readers.

### *Needs work* :thumbsdown:

* Code annotation style used interrupts the flow of the code, and makes your code hard to read (You have sandwiched lines of comments in between piped code and different arguments in the same command). Readers are now forced to scan through comments to get at the code. If you think you need to use a lot of comments to explain some code it might be a good idea to break the code up into several code chunks, or  you can place the comments on the side, so that if the reader needs a better explanation of the code than they can read the comments. Vincenzo made the comment about how having comments on the side make the code long, but your comments should be brief and simple.

* The length of your tables and the use of `glimpse()`, `head/tail()` make it hard to follow how each read/write command is working. It would be more effective to show how these work with a small easy to view data frame/table! 

# **Graphs**

### *The good* :thumbsup:

* Deep dive into theme!

* ploties look dope!!

### *Needs work* :thumbsdown:

* Some of your x-axis text is overlapping. Try adding this to your ggplot code to rotate longer labels:
```
theme( axis.text.x = element_text( angle = 90 ) )
```

* Please add a title to each graph.

*Please insure that all x and y axes have been refurbished for better quality graphics! For example pop is not a refurbished axes title.

# **Tables** 

### *The good* :thumbsup:

* Good use of kable to render tables!

### *Needs work* :thumbsdown:

* Is using head the most effective way to show a reader a representative dataset? For example using head may confuse the reader and result in them thinking that Afghanistan is the only country in the dataset (if they haven’t seen the dataset before). If you would like to show the readers a more representative sample of the dataset your using you can do a stratified random sample some from each group/continent…. Try using:

```
DATA %>% 
  group_by( `the groups you want` ) %>% 
  sample_frac() %>%  #or sample_n
  kable()
```

# **Creativity & Ease of Access**

### *The good* :thumbsup:

* Excellent use of TOC

* Easy to navigate from canvas to your github, and easy to navigate your repo once I was there.

* Above and beyond exploration in every section!

* Deep dive into forcats!!

### *Needs work* :thumbsdown:

* Your main repo has a lot of unsorted data files, which will overwhelm readers as the repo grows! For next time please place these datasets into their own sub folder just like you did for your figures.

**Comment by zjbthomas** [comment]
Hi @ChadFibke , thanks for your comment! It is the longest TA comment I got right now :stuck_out_tongue_closed_eyes: , with a lot of helpful tips!

- For sure I will try to make better names for my saved figures, and try to organize all files in one folder. Sorry that I forget to do this for some of the files.
- I also have the feeling that using `glimpse()`, `head()`, and `tail()` are not the best ways to show previews to readers. I will try to use `sample_frac()` as you suggest and see if it performs better.
- I am sure I make titles for each graph, are they not clear to readers? Could you please explain more about your feelings on them so I can try to improve?
- The only figure that I use *pop* as in x-axis is because it is directly copied from homework 01 :stuck_out_tongue_closed_eyes: , and, as the requirement of the homework, I tried to improve it. So actually it is a good example to support your advice.
- Thanks for showing me the usage of `angle = 90`.

Thank you.

**Comment by ChadFibke** [comment]
Hey @zjbthomas, No problem!

* You are totally right about the titles comment, my mistake. I will update your mark ASAP. 

* Sorry the pop was a bad example to use! For the "refurbished axes title" I mean you should capitalize every axis you make. If you look at your first couple of graphs you see country, but then you change it to **C**ountry in the later half of your assignment.

**Comment by zjbthomas** [comment]
> * Sorry the pop was a bad example to use! For the "refurbished axes title" I mean you should capitalize every axis you make. If you look at your first couple of graphs you see country, but then you change it to **C**ountry in the later half of your assignment.

Got it, thanks!
